File: EnumList.ts
// @author: 321_Closet
/* 
    @description: a multi-use library written in typescript
    indented for use inside the berry game framework for ROBLOX
*/

const MAX_LIST_LENGTH = 10;

export class c_EnumList {
	public list;
	public l_name;
	constructor(name: string, array: []) {
		this.l_name = name;
		assert(array.length > MAX_LIST_LENGTH, "Enum Lists may not exceed threshold of 10 strings");
		this.list = array;
		table.freeze(this.list);
	}

	/**
	 * @method BelongsTo: Ask if a certain const char is a member of a list
	 * in this case the list object of which the method is called
	 */
	public BelongsTo(obj: string) {
		let i = 0;
		for (i = 0; i <= this.list.length; i++) {
			if (this.list[i] === obj) {
				return true;
			} else {
				return false;
			}
		}
	}

	/**
	 * @method GetEnumItems: Return a read-only array of the objects enum items
	 */
	public GetEnumItems(): [] {
		return this.list;
	}

	/**
	 * @method GetName: Return the name of the enum list
	 */
	public GetName() {
		return this.l_name;
	}
}

declare module "C:/berry/src/EnumList";

File: EnumList.lua
-- Compiled with roblox-ts v3.0.0
-- @author: 321_Closet
--[[
	 
	    @description: a multi-use library written in typescript
	    indented for use inside the berry game framework for ROBLOX
	
]]
local MAX_LIST_LENGTH = 10
local c_EnumList
do
	c_EnumList = setmetatable({}, {
		__tostring = function()
			return "c_EnumList"
		end,
	})
	c_EnumList.__index = c_EnumList
	function c_EnumList.new(...)
		local self = setmetatable({}, c_EnumList)
		return self:constructor(...) or self
	end
	function c_EnumList:constructor(name, array)
		self.l_name = name
		local _arg0 = array.length > MAX_LIST_LENGTH
		assert(_arg0, "Enum Lists may not exceed threshold of 10 strings")
		self.list = array
		table.freeze(self.list)
	end
	function c_EnumList:BelongsTo(obj)
		local i = 0
		do
			i = 0
			local _shouldIncrement = false
			while true do
				if _shouldIncrement then
					i += 1
				else
					_shouldIncrement = true
				end
				if not (i <= self.list.length) then
					break
				end
				if self.list[i + 1] == obj then
					return true
				else
					return false
				end
			end
		end
	end
	function c_EnumList:GetEnumItems()
		return self.list
	end
	function c_EnumList:GetName()
		return self.l_name
	end
end
return {
	c_EnumList = c_EnumList,
}
